src/api/hooks/channel.ts

```diff
@@ -121,3 +125,43 @@ export const useUnfollowChannel = (opts?: UnfollowChannelOpts) => {
     ...rest,
   }
 }
+
+type MostPopularChannelsOpts = QueryHookOptions<GetMostViewedChannelsQuery>
+export const useMostViewedChannelsIds = (
+  variables?: GetMostViewedChannelsQueryVariables,
+  opts?: MostPopularChannelsOpts
+) => {
+  const { data, ...rest } = useGetMostViewedChannelsQuery({ ...opts, variables })
+  return {
+    mostViewedChannels: data?.mostViewedChannels,
+    ...rest,
+  }
+}
+
+export const useMostViewedChannels = (
+  variables?: GetMostViewedChannelsQueryVariables,
+  opts?: MostPopularChannelsOpts
+) => {
+  const { mostViewedChannels } = useMostViewedChannelsIds(variables, opts)
+
+  const mostViewedChannelsIds = useMemo(() => {
+    if (mostViewedChannels) {
+      return mostViewedChannels.map((item) => item.id)
+    }
+    return null
+  }, [mostViewedChannels])
+
+  const { channels, ...rest } = useChannels(
+    {
+      where: {
+        id_in: mostViewedChannelsIds,
+      },
+    },
+    { skip: !mostViewedChannelsIds }
+  )
+
+  return {
+    channels,
+    ...rest,
+  }
+}
```

Comment:
I wonder, do we need to export both queries, wouldn't be enough with `useMostViewedChannels` for every use case?

Reply:
The answer can be found in `InfiniteVideoGrid`. To use most popular videos in infinite grid we need only id's. And `useMostViewedChannels` is for the cases we won't need pagination. Check `InfiniteVideoGrid` and I'm sure you'll understand.

Reply:
okay makes sense 👍